src/components/GoalPage/GoalPage.tsx

```diff
@@ -191,6 +193,36 @@ export const GoalPage = ({ user, ssrTime, params: { id } }: ExternalPageProps<{
         [goal, invalidateFn, addParticipantMutation],
     );
 
+    const onAssigneeChange = useCallback(
+        async (activity?: ActivityByIdReturnType) => {
+            if (goal && activity?.user) {
+                setCurrentOwner(activity.user);
+                try {
+                    await update({
+                        ...(goal as unknown as GoalUpdate),
+                        owner: {
+                            id: activity?.id || '',
+                            user: {
+                                ...activity?.user,
+                            },
+                        },
+                        // Need to pad fields due to zod validation
+                        parent: {
+                            id: goal.project?.id || '',
+                            title: goal.project?.title || '',
+                            flowId: goal.project?.flowId || '',
+                        },
+                        estimate: goal._lastEstimate,
+                    });
+                } catch (error: any) {
+                    setCurrentOwner(currentOwner);
+                }
+                invalidateFn();
```

Comment:
Why this is not in `try` condition? Currently u invalidate data in any case of update result.